src/Button/Button.js

```diff
@@ -71,34 +80,96 @@ const StyledButton = createComponent({
       appearance: none;
       border-radius: ${borderRadius}px;
       pointer-events: ${disabled ? 'none' : 'auto'};
-      opacity: ${disabled ? 0.65 : 1};
       color: ${outline ? backgroundColor : fontColor};
       height: ${height}px;
-      padding: 0 ${height * 0.5}px;
+      padding: 0 16px;
       font-size: ${fontSize}px;
       width: ${block ? '100%' : 'auto'};
-      background: ${outline || transparent ? 'transparent' : backgroundColor};
-      border: ${transparent ? 'none' : `1px solid ${backgroundColor}`};
+      background: ${outline ? 'transparent' : backgroundColor};
+      border-color: ${outline ? backgroundColor : borderColor || backgroundColor};
+      border-style: solid;
+      border-width: 1px;
       transition: 175ms;
+      white-space: nowrap;
+      outline: none;
+      user-select: none;
 
-      ${loading && loadingCss({ height, fontColor, outline, backgroundColor })};
+      ${icon &&
+        hasText &&
+        css`
+          padding-left: 8px;
+
+          i {
+            padding-right: 4px;
+          }
+        `}
+
+      ${loading && loadingCss({ height, fontColor, backgroundColor, borderColor, outline })};
 
       &:hover {
-        background: ${outline ? 'transparent' : lighten(0.05, backgroundColor)};
-        border-color: ${lighten(0.05, backgroundColor)};
+        background-color: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${hover &&
+          css`
+            background-color: ${hover.backgroundColor || backgroundColor};
+            border-color: ${hover.borderColor || hover.backgroundColor || backgroundColor};
+            color: ${hover.fontColor || fontColor};
```

Comment:
When hovering over a text only button, I'm not seeing the text color change per the Figma specs. Same things goes for the `active` state.